checker/jdk/nullness/src/javax/script/AbstractScriptEngine.java

```diff
@@ -0,0 +1,313 @@
+import java.util.Iterator;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Provides a standard implementation for several of the variants of the <code>eval</code>
+ * method.
+ * <br><br>
+ * <code><b>eval(Reader)</b></code><p><code><b>eval(String)</b></code><p>
+ * <code><b>eval(String, Bindings)</b></code><p><code><b>eval(Reader, Bindings)</b></code>
+ * <br><br> are implemented using the abstract methods
+ * <br><br>
+ * <code><b>eval(Reader,ScriptContext)</b></code> or
+ * <code><b>eval(String, ScriptContext)</b></code>
+ * <br><br>
+ * with a <code>SimpleScriptContext</code>.
+ * <br><br>
+ * A <code>SimpleScriptContext</code> is used as the default <code>ScriptContext</code>
+ * of the <code>AbstractScriptEngine</code>..
+ *
+ * @author Mike Grogan
+ * @since 1.6
+ */
+public abstract class AbstractScriptEngine  implements ScriptEngine {
+
+    /**
+     * The default <code>ScriptContext</code> of this <code>AbstractScriptEngine</code>.
+     */
+
+    protected ScriptContext context;
+
+    /**
+     * Creates a new instance of AbstractScriptEngine using a <code>SimpleScriptContext</code>
+     * as its default <code>ScriptContext</code>.
+     */
+    public AbstractScriptEngine() {
+
+        context = new SimpleScriptContext();
+
+    }
+
+    /**
+     * Creates a new instance using the specified <code>Bindings</code> as the
+     * <code>ENGINE_SCOPE</code> <code>Bindings</code> in the protected <code>context</code> field.
+     *
+     * @param n The specified <code>Bindings</code>.
+     * @throws NullPointerException if n is null.
+     */
+    public AbstractScriptEngine(Bindings n) {
+
+        this();
+        if (n == null) {
+            throw new NullPointerException("n is null");
+        }
+        context.setBindings(n, ScriptContext.ENGINE_SCOPE);
+    }
+
+    /**
+     * Sets the value of the protected <code>context</code> field to the specified
+     * <code>ScriptContext</code>.
+     *
+     * @param ctxt The specified <code>ScriptContext</code>.
+     * @throws NullPointerException if ctxt is null.
+     */
+    public void setContext(ScriptContext ctxt) {
+        if (ctxt == null) {
+            throw new NullPointerException("null context");
+        }
+        context = ctxt;
+    }
+
+    /**
+     * Returns the value of the protected <code>context</code> field.
+     *
+     * @return The value of the protected <code>context</code> field.
+     */
+    public ScriptContext getContext() {
+        return context;
+    }
+
+    /**
+     * Returns the <code>Bindings</code> with the specified scope value in
+     * the protected <code>context</code> field.
+     *
+     * @param scope The specified scope
+     *
+     * @return The corresponding <code>Bindings</code>.
+     *
+     * @throws IllegalArgumentException if the value of scope is
+     * invalid for the type the protected <code>context</code> field.
+     */
+    public @Nullable Bindings getBindings(int scope) {
```

Comment:
How do you know this can be null?  It looks like it from the code, but the documentation does not indicate it.

Reply:
Wrongly annotated. Have updated it.
-Thanks!